plugins/by-name/dap-python/default.nix

```diff
@@ -0,0 +1,94 @@
+{
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib)
+    optionalString
+    types
+    ;
+  inherit (lib.nixvim)
+    defaultNullOpts
+    mkNullOrOption
+    toLuaObject
+    ;
+
+  dapHelpers = import ../dap/dapHelpers.nix { inherit lib; };
+in
+lib.nixvim.plugins.mkNeovimPlugin {
+  name = "dap-python";
+  package = "nvim-dap-python";
+
+  maintainers = [ lib.maintainers.khaneliman ];
+
+  extraOptions = {
+    adapterPythonPath = lib.mkOption {
+      default = lib.getExe (pkgs.python3.withPackages (ps: with ps; [ debugpy ]));
+      defaultText = lib.literalExpression ''
+        lib.getExe (pkgs.python3.withPackages (ps: with ps; [ debugpy ]))
+      '';
+      description = "Path to the python interpreter. Path must be absolute or in $PATH and needs to have the debugpy package installed.";
+      type = types.str;
+    };
+
+    customConfigurations = mkNullOrOption (types.listOf dapHelpers.configurationOption) "Custom python configurations for dap.";
+
+    resolvePython = defaultNullOpts.mkLuaFn null ''
+      Function to resolve path to python to use for program or test execution.
+      By default the `VIRTUAL_ENV` and `CONDA_PREFIX` environment variables are used if present.
+    '';
+
+    testRunner = mkNullOrOption (types.either types.str types.rawLua) ''
+      The name of test runner to use by default.
+      The default value is dynamic and depends on `pytest.ini` or `manage.py` markers.
+      If neither is found "unittest" is used.
+    '';
+
+    testRunners = mkNullOrOption (with lib.types; attrsOf strLuaFn) ''
+      Set to register test runners.
+      Built-in are test runners for unittest, pytest and django.
+      The key is the test runner name, the value a function to generate the
+      module name to run and its arguments.
+      See |dap-python.TestRunner|.
+    '';
```

Comment:
Feels like this description would benefit from some line breaks?

Also, lets add some `` ` `` formatting to `|dap-python.TestRunner|`